test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
+        @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
+        @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
+        @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
+        @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
+
+        a = speye(5) + triu(0.1*sprandn(5, 5, 0.2))
+        b = randn(5,3) + im*randn(5,3)
+        @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
+        @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
+        @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
+        @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
+        @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
+        @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
+
+        a = spdiagm(randn(5)) + im*spdiagm(randn(5))
+        b = randn(5,3)
+        @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
+        @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
+        @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
+        @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
+        @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
+        @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
+
+        b = randn(5,3) + im*randn(5,3)
+        @test (maximum(abs.(a*b - Array(a)*b)) < 100*eps())
+        @test (maximum(abs.(a'b - Array(a)'b)) < 100*eps())
+        @test (maximum(abs.(a.'b - Array(a).'b)) < 100*eps())
+        @test (maximum(abs.(a\b - Array(a)\b)) < 1000*eps())
+        @test (maximum(abs.(a'\b - Array(a')\b)) < 1000*eps())
+        @test (maximum(abs.(a.'\b - Array(a.')\b)) < 1000*eps())
+    end
+    end
+end
+
+@testset "Matrix multiplication and kron" begin
+    for i = 1:5
+        a = sprand(10, 5, 0.7)
+        b = sprand(5, 15, 0.3)
+        @test maximum(abs.(a*b - Array(a)*Array(b))) < 100*eps()
+        @test maximum(abs.(Base.SparseArrays.spmatmul(a,b,sortindices=:sortcols) - Array(a)*Array(b))) < 100*eps()
+        @test maximum(abs.(Base.SparseArrays.spmatmul(a,b,sortindices=:doubletranspose) - Array(a)*Array(b))) < 100*eps()
+        @test Array(kron(a,b)) == kron(Array(a), Array(b))
+        @test Array(kron(Array(a),b)) == kron(Array(a), Array(b))
+        @test Array(kron(a,Array(b))) == kron(Array(a), Array(b))
+        c = sparse(rand(Float32,5,5))
+        d = sparse(rand(Float64,5,5))
+        @test Array(kron(c,d)) == kron(Array(c),Array(d))
+        f = Diagonal(rand(5))
+        @test Array(a*f) == Array(a)*f
+        @test Array(f*b) == f*Array(b)
+    end
+end
+
 sA = sprandn(3, 7, 0.5)
 sC = similar(sA)
 dA = Array(sA)
 b = randn(7)
-@test dA * Diagonal(b) == sA * Diagonal(b)
-@test dA * Diagonal(b) == scale!(sC, sA, b)
-@test dA * Diagonal(b) == scale!(copy(sA), b)
-b = randn(3)
-@test Diagonal(b) * dA == Diagonal(b) * sA
-@test Diagonal(b) * dA == scale!(sC, b, sA)
-@test Diagonal(b) * dA == scale!(b, copy(sA))
-
-@test dA * 0.5            == sA * 0.5
-@test dA * 0.5            == scale!(sC, sA, 0.5)
```

Comment:
Perhaps just "getindex bounds checking"? The bounds checking itself is not random, just the test matrix :).